### PR TITLE
Fixed crashing of storage app

### PR DESCRIPTION
**Pull request by maneesht**
Tested on Android L (API 22) and Android 12L (API 32)

### CODE REVIEW

File: storage/app/src/main/java/com/google/firebase/quickstart/firebasestorage/java/MainActivity.java [hunk 1]
```diff
@@ -85,6 +87,15 @@ protected void onCreate(Bundle savedInstanceState) {
         }
         onNewIntent(getIntent());
 
+        this.intentLauncher = registerForActivityResult(
```
Comment (thatfiredev) [inline storage/app/src/main/java/com/google/firebase/quickstart/firebasestorage/java/MainActivity.java:90]: nit: is `this` really necessary? I would probably omit it for simplicity :)

File: storage/app/src/main/java/com/google/firebase/quickstart/firebasestorage/java/MainActivity.java [hunk 2]
```diff
@@ -193,15 +204,7 @@ private void launchCamera() {
...
-        ActivityResultLauncher<String[]> intentLauncher = registerForActivityResult(
-                new ActivityResultContracts.OpenDocument(), fileUri -> {
-                    if (fileUri != null) {
-                        uploadFromUri(fileUri);
-                    } else {
-                        Log.w(TAG, "File URI is null");
-                    }
-                });
-        intentLauncher.launch(new String[]{ "image/*" });
+        this.intentLauncher.launch(new String[]{ "image/*" });
```
Comment (thatfiredev) [inline storage/app/src/main/java/com/google/firebase/quickstart/firebasestorage/java/MainActivity.java:207]: Same comment as above
```suggestion
       intentLauncher.launch(new String[]{ "image/*" });
```

File: storage/app/src/main/java/com/google/firebase/quickstart/firebasestorage/kotlin/MainActivity.kt [hunk 3]
```diff
@@ -159,14 +170,7 @@ class MainActivity : AppCompatActivity(), View.OnClickListener {
...
         // Pick an image from storage
-        val intentLauncher = registerForActivityResult(ActivityResultContracts.OpenDocument()) { fileUri ->
-            if (fileUri != null) {
-                uploadFromUri(fileUri)
-            } else {
-                Log.w(TAG, "File URI is null")
-            }
-        }
-        intentLauncher.launch(arrayOf("image/*"))
+        this.cameraIntent.launch(arrayOf("image/*"))
```
Comment (thatfiredev) [inline storage/app/src/main/java/com/google/firebase/quickstart/firebasestorage/kotlin/MainActivity.kt:173]: Same comment as above.
```suggestion
        cameraIntent.launch(arrayOf("image/*"))
```

**Review by thatfiredev — COMMENTED** [review]
Thanks @maneesht !
The changes LGTM, I just have a minor comment.

File: storage/app/build.gradle [hunk 4]
```diff
@@ -53,6 +53,7 @@ dependencies {
 
     implementation 'androidx.activity:activity-ktx:1.6.1'
     implementation 'androidx.appcompat:appcompat:1.5.1'
+    implementation 'com.android.databinding:viewbinding:7.3.1'
```
[the remark this answers is not in the document]
Comment (maneesht) [reply storage/app/build.gradle:56]: I did have an issue locally, but I'm afraid if I don't commit this, others will have the same issue with their env

[the remark this answers is not in the document]
Comment (maneesht) [reply storage/app/build.gradle]: Weird, I rebuilt and saw viewbinding errors, and then added it, and they went away. Now I removed the dep import and no error still. Not sure what happened...

[the remark this answers is not in the document]
Comment (thatfiredev) [reply storage/app/build.gradle]: Yes, viewbinding sometimes behaves weirdly, I kinda got used to seeing these errors.

**Pull request merged**